src/components/chat/comment-card.tsx

```diff
@@ -66,19 +66,22 @@ export const CommentCard: React.FC<IProps> = ({ activeNavTab, user, postedCommen
             const userInitialBackgroundColor = ["#f79999", "#ffc18a", "#99d099", "#ff9", "#b2b2ff", "#efa6ef"];
             const commenterInitial = comment.name.charAt(0);
             const userInitialBackgroundColorIndex = parseInt(comment.uid, 10) % 6;
-            const isCurrentUserComment = user?.id === comment.uid;
-            const backgroundStyle = isCurrentUserComment
+            const isOwnComment = user?.id === comment.uid;
+            const shouldShowUserIcon = isOwnComment;
+            // can't delete comment until we have a valid server-generated id
+            const shouldShowDeleteIcon = isOwnComment && !comment.id.startsWith("pending-");
```

Comment:
This seems a little hacky, IMO (i.e., encoding the validity/persistence of the comment by adding the 'pending-' prefix to the ID string). I'm wondering if it would be better to have an explicit boolean for any data element this is not (yet) finalized (i.e., in the backing store). But maybe we use this pattern elsewhere, so it's a de-facto convention?

Reply:
The only difference between a fully vested comment and a pending comment is the id. Marking the id as `pending` seems like a natural way of indicating that. I'm not sure I see what the advantage of a more complicated implementation that introduced a new type and/or a new optional property would be.